lib/manager/input_dispatcher.dart

```diff
@@ -24,11 +24,13 @@ class InputDispatcher {
     for (var handler in _handlers) {
       final response = handler.handleInput(event);
       if (response.handled) {
-        for (final component in response.dirty ?? []) {
-          _renderManager.markDirty(component);
-        }
+        if (!_renderManager.needsRecompute) {
+          for (final component in response.dirty ?? []) {
```

Comment:
Shouldn't response.dirty always be a list (not nullable) as if there aren't any dirty components, the list will just be empty.